Map/RoomBase.gd

```diff
@@ -0,0 +1,10 @@
+extends Resource
+class_name RoomBase
+## Class for a room on the map. Provides base classes for all rooms to use regardless of the type
+##
+## Holds basic functionality for a room on the map
+
+@export var possible_events: Array = [EventHeal, EventMob, EventRandom, EventShop]
+@export var light_level: int = 0
+@export var has_torch: bool = false
+@export var this_room_event: EventBase
```

Comment:
If we're going to be making several instances of RoomBase, I don't think it's necessary to prepend `this` for `this_room_event`. 🤔 It shouldn't have knowledge of other rooms in this class so we should be able to just name it `event` or `room_event`!

Reply:
`room_event` seems like a good name